src/controllers/Controller.js

```diff
@@ -262,23 +263,167 @@ async loginRestaurant(req, res) {
-                const data = notifications.length ? notifications : `No notifications found for RestaurantID: ${restaurant_id}`;
+                const orders = await dbRepo.getOrdersByRestaurantID(restaurantID);
+                const data = orders.length ? orders : `No orders found for Restaurant ID: ${restaurantID}`;
                 response(res, { data });
             }
         });
+    } catch (error) {
+        response(res, { status: 400, data: error.message });
+    }
+  }
+
+
+  async respondToFeedback(req, res) {
+    try {
+        const { restaurantID } = req.params;
+        const body = await getPostBodyAsync(req);
+        const { reviewID, response: feedbackResponse } = body;
+        const token = req.headers.authorization;
+
+        jwt.verify(token, secretKey, async (err, decoded) => {
+            if (err) {
+                response(res, { status: 401, data: { message: 'Unauthorized' } });
+            } else {
+                const result = await dbRepo.respondToReview(reviewID, restaurantID, feedbackResponse);
+                if (result) {
+                    response(res, { data: "Response added successfully!" });
+                } else {
+                    throw new Error("Failed to add response.");
+                }
+            }
+        });
+    } catch (error) {
+        response(res, { status: 400, data: { message: error.message } });
+    }
+  }
+
+  async addMenuItem(req, res) {
+    try {
+        const { restaurantID } = req.params;
+        const token = req.headers.authorization;
+        const body = await getPostBodyAsync(req);
+        jwt.verify(token, secretKey, async (err, decoded) => {
+            if (err) {
+                response(res, { status: 401, data: { message: 'Unauthorized' } });
+            } else {
+                const {
+                    item_name,
+                    description,
+                    price,
+                    spice_level,
+                    is_available,
+                    category,
+                    image_url,
+                    is_featured
+                } = body;
+
+                await dbRepo.addMenuItemToMenu({
+                    restaurantID,
+                    item_name,
+                    description,
+                    price,
+                    spice_level,
+                    is_available,
+                    category,
+                    image_url,
+                    is_featured
+                });
+
+                response(res, { data: { message: 'Menu item added successfully.' } });
+            }
+        });
     } catch (error) {
         response(res, { status: 400, data: { message: error.message } });
     }
 }
 
+
+  async postRestaurantReview(req, res) {
+    try {
+        const { restaurantID } = req.params;
+        const token = req.headers.authorization;
+        const body = await getPostBodyAsync(req);
```

Comment:
Please check the flow for POST requests in the routes, do not use the getPostBodyAsync method over here, but rather in a separate validation script defined for this method. Check the other POST requests in the routes.

Reply:
Kindly verify if its fine now. I have titled it `validatePostRequests` since it is a generic function for any POST body.